src/borrow_tracker/tree_borrows/unimap.rs

```diff
@@ -36,13 +36,42 @@ pub struct UniKeyMap<K> {
 }
 
 /// From UniIndex to V
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, Eq)]
 pub struct UniValMap<V> {
     /// The mapping data. Thanks to Vec we get both fast accesses, and
     /// a memory-optimal representation if there are few deletions.
     data: Vec<Option<V>>,
 }
 
+impl<V: PartialEq> UniValMap<V> {
+    /// Exact equality of two maps.
+    /// Less accurate but faster than `equivalent`, mostly because
+    /// of the fast path when the lengths are different.
+    pub fn identical(&self, other: &Self) -> bool {
+        self.data == other.data
+    }
+
+    /// Equality up to trailing `None`s of two maps, i.e.
+    /// do they represent the same mapping ?
+    pub fn equivalent(&self, other: &Self) -> bool {
+        let min_len = self.data.len().min(other.data.len());
+        self.data[min_len..].iter().all(Option::is_none)
+            && other.data[min_len..].iter().all(Option::is_none)
+            && (self.data[..min_len] == other.data[..min_len])
+    }
+}
+
+impl<V: PartialEq> PartialEq for UniValMap<V> {
+    /// We found that using `equivalent` rather than `identical`
```

Comment:
```suggestion
    /// 2023-05: We found that using `equivalent` rather than `identical`
```
It helps to date such comments